classes/Hook/HookDisplayFooter.php

```diff
@@ -0,0 +1,110 @@
+<?php
+/**
+ * 2007-2020 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\Ps_Googleanalytics\Hooks;
+
+use Context;
+use Hook;
+use PrestaShop\Module\Ps_Googleanalytics\GoogleAnalyticsTools;
+use PrestaShop\Module\Ps_Googleanalytics\Handler\GanalyticsDataHandler;
+use PrestaShop\Module\Ps_Googleanalytics\Handler\GanalyticsJsHandler;
+use PrestaShop\Module\Ps_Googleanalytics\Wrapper\ProductWrapper;
+use Ps_Googleanalytics;
+use Tools;
+
+class HookDisplayFooter implements HookInterface
+{
+    private $module;
+    private $context;
+
+    public function __construct(Ps_Googleanalytics $module, Context $context)
+    {
+        $this->module = $module;
+        $this->context = $context;
+    }
+
+    /**
+     * run
+     *
+     * @return string
+     */
+    public function run()
+    {
+        $gaTools = new GoogleAnalyticsTools();
+        $gaTagHandler = new GanalyticsJsHandler($this->module, $this->context);
+        $ganalyticsDataHandler = new GanalyticsDataHandler(
+            $this->context->cart->id,
+            $this->context->shop->id
+        );
+
+        $gaScripts = '';
+        $this->module->js_state = 0;
+        $gacarts = $ganalyticsDataHandler->manageData('', 'R');
+        $controller_name = Tools::getValue('controller');
+
+        if (count($gacarts) > 0 && $controller_name != 'product') {
+            $this->module->filterable = 0;
+
+            foreach ($gacarts as $gacart) {
+                if (isset($gacart['quantity'])) {
+                    if ($gacart['quantity'] > 0) {
+                        $gaScripts .= 'MBG.addToCart(' . json_encode($gacart) . ');';
```

Comment:
In an ideal world (but this would be a huge work) it would be great to have something like a `MBGScriptManager` class able to do all of these string concatenations.

So
`$gaScripts .= 'MBG.addToCart(' . json_encode($gacart) . ');';`
would become
`$MBGScriptManager->writeAddToCartStatement($gaScripts, $gaCart));`

And you could even unit-test it

Might be a good candidate for future release ?

Reply:
This would allow some cool features like
- avoiding duplicate `MBG.addToCart()` statements
- having functions like `MBGScriptManager->getAddToCartLines()`

And it would be easier to debug MBG statements

Reply:
I totally agree with that. But the first action will to understand exactly what are those MBG scripts and how it works. Currently it's a black box for me

Reply:
<img src="https://cdn3.vectorstock.com/i/1000x1000/23/37/black-box-with-magic-lights-vector-6472337.jpg>